### ISSUE TITLE
Todo List

### ISSUE THREAD

**Issue by kyranstar** [kyranstar/Hunt-The-Wumpus#8]
# TODO:
 - [ ] Be able to receive inputs
 - [ ] Be able to move the player from room to room (WASD?)
 - [ ] Validate that the player doesn't move out of bounds
 - [ ] Move the viewport based on the players location (center the player in the screen)
 - [ ] Make the rooms pretty (pink is so ugly :grin:)

**Comment by WasabiFan** [comment]
@kyranstar What happened today? Were you able to demo anything?

> Make the rooms pretty (pink is so ugly :grin:)

I didn't intend for it to be pretty ;) I just picked an image from my "Pictures" folder.

**Comment by kyranstar** [comment]
@WasabiFan We basically showed them what we had, and they thought that we were pretty "ahead of the curve." We didn't really get much of anything done besides talking about what we had to do for next week.

**Comment by kyranstar** [comment]
By the way, (for everybody) besides unit tests, [Debug.Assert](http://stackoverflow.com/questions/163538/c-sharp-what-does-the-assert-method-do-is-it-still-useful) is pretty useful.

**Comment by WasabiFan** [comment]
## Detailed TODO (longer-term)
- [ ] Add high score object
  - [ ] Give it an API to be able to read and write scores to be stored in volatile memory
    - Username
    - Score
    - Other info?
  - [ ] Save high scores locally (decide where we want to save it first)
  - [ ] Look into saving them to an Azure backend (for brownie points)
- [ ] Figure out what trivia is supposed to do
  - Does it present information in every new room, or ask the user a trivia question in every room?
  - If it *does* give them the answers beforehand, can we defer displaying that info until just before they have to answer a question (a "review" period?)
- [ ] Add trivia object
  - [ ] Make API to get the trivia info (at least from a hard-coded array)
    - Get the question, a hint or two if needed, maybe some metrics
    - Need to figure out where we want to store the info on the current trivia question
  - [ ] Load the trivia from a file (or wherever we decide to store it)
  - [ ] Add extra info API regarding which questions they have gotten wrong, which ones have been displayed, etc.
  - [ ] Think about another mini-game in place of trivia (as allowed in the spec)
- [ ] Implement the rest of the Wumpus tracking
  - [ ] Position
  - [ ] Other movement/actions?
- [ ] Add player warning detection
- [ ] Add bats and other hazards
- [ ] Add all map/HUD objects to renderer
  - [ ] Wumpus
  - [ ] Bats
  - [ ] Pits
  - [ ] Inventory
    - [ ] Gold
    - [ ] Arrows
  - [ ] Trivia
- [ ] Add inventory management
  - [ ] Gold pickup
  - [ ] Arrow pickup

I'm sure that there are more things, but this should be a good list of the basic features that we need. Are there any big ones that I missed? Feel free to edit my comment if needed.

**Comment by BlueRosesSonyPony** [comment]
I can't figure out where/how to add the trivia class, and I can't see what you have working so far because of the errors popping up, so I'll bring in this laptop to school tomorrow and if one of you could please show me how to get started that would be much appreciated. Thanks.

**Comment by WasabiFan** [comment]
Sounds good -- I'm sure we can help you figure it out.

**Comment by WasabiFan** [comment]
@BlueRosesSonyPony I just realized that we forgot to see if we could get your PC set up in class -- sorry! Remind me in class tomorrow.

**Comment by kyranstar** [comment]
@WasabiFan I talked to her today, and I think we figured it out so that she could commit. We'll see if there are any issues.

**Comment by WasabiFan** [comment]
Ahhhh I see, thanks.

**Comment by kyranstar** [comment]
TODO (edit this list)

- [ ] Trivia questions and hints to show up in UI
- [ ] Hazards handled (warnings and hitting detection)
- [ ] Shooting the wumpus
- [ ] Game over
- [ ] Main menu buttons

- [ ] Command line stuff (display warnings, display all tiles, etc.)

**Comment by WasabiFan** [comment]
@BlueRosesSonyPony If you get a chance today, can you start trying to implement the trivia setup that we discussed on Friday? I think that if we can get that framework in place soon everything else will just be a matter of one-line calls in the appropriate places.

**Comment by BlueRosesSonyPony** [comment]
Sorry, just saw this today -- will work on it either later today or
tomorrow.

**Comment by BlueRosesSonyPony** [comment]
Is anyone else getting an error when they try to run the solution?

**Comment by nikbyk** [comment]
What error?





        Sent by Outlook for Android




On Tue, May 5, 2015 at 8:08 PM -0700, "BlueRosesSonyPony" <notifications@github.com> wrote:
Is anyone else getting an error when they try to run the solution?

On Tue, May 5, 2015 at 2:54 PM, Sonia Skarbek <sonia.skarbek@gmail.com>
wrote:

> Sorry, just saw this today -- will work on it either later today or
> tomorrow.
>
> On Sun, May 3, 2015 at 6:18 PM, Wasabi Fan <notifications@github.com>
> wrote:
>
>> @BlueRosesSonyPony <https://github.com/BlueRosesSonyPony> If you get a
>> chance today, can you start trying to implement the trivia setup that we
>> discussed on Friday? I think that if we can get that framework in place
>> soon everything else will just be a matter of one-line calls in the
>> appropriate places.
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/kyranstar/Hunt-The-Wumpus/issues/8#issuecomment-98563664>
>> .
>>
>
>


---

**Comment by BlueRosesSonyPony** [comment]
Name: Engine
Value: The value of the local or argument 'Engine' is unobtainable at this
time.
Type: EmptyKeys.UserInterface.Engine

**Comment by WasabiFan** [comment]
Is this showing up as an exception (in a separate box while the application is running) or as a compile error (in the "errors" panel at the bottom of the screen)? Can you include a screenshot of the situation just so that I can get some context?

**Comment by BlueRosesSonyPony** [comment]
It's a compile error.

**Comment by WasabiFan** [comment]
In what file and on what line is this occurring?

**Comment by WasabiFan** [comment]
Oh, wait... I think I might know what's going on. By any chance were you looking at the "Watch 1" panel when you copied that error info? Errors that show up there are not going to prevent you from compiling or running -- it just means that at some point someone added some debug code to your Visual Studio environment. You can safely ignore this.

Have you actually tried to run the project using the "Start" button at the top of the window?

**Comment by BlueRosesSonyPony** [comment]
I was using start, but I never continued all the way through because if the error, it might just be a debug thing, I'll continue all the way through next time and see what happens.

**Comment by WasabiFan** [comment]
OK, sounds good.

If it shows you a dialog box that says "There were build errors. Would you like to continue and run the last successful build?", answer "no" and post back the error that appears in the "Error list" section.

If it shows something like this:
![image](https://community.devexpress.com/blogs/markmiller/VisualStudioExceptionHelperDialog_27401710.png)

Click the "Copy exception detail to the clipboard" link and post that copied text here so that we can try to help you fix it.

**Comment by WasabiFan** [comment]
@nikbyk Can you remove the margin on the character in the same way you did for the coin?

**Comment by nikbyk** [comment]
Will do





        Sent by Outlook for Android




On Fri, May 8, 2015 at 8:24 AM -0700, "Wasabi Fan" <notifications@github.com> wrote:
@nikbyk Can you remove the margin on the character in the same way you did for the coin?

---

**Comment by WasabiFan** [comment]
@BlueRosesSonyPony What's the status of those extra tracking properties that we discussed today? If you are able to finish that and commit/push it today I should be able to get the overall trivia control done before class tomorrow. I already have the GUI part done, so it's just that tracking that we need to be pretty much feature-complete.

**Comment by BlueRosesSonyPony** [comment]
I was preoccupied earlier - I'll do it now.

**Comment by BlueRosesSonyPony** [comment]
There I changed it. Not sure if it's quite what you wanted though - you can change it however you want.

**Comment by WasabiFan** [comment]
Yup -- looks pretty close. There's a small change that I'll need to make, but I think this it all we'll need.

> you can change it however you want

Yeah, I understand :wink: I've been trying to stay within my "graphics" domain and just fill in the cracks to give everyone else a chance to work on their parts. But now that you have implemented the bulk of the class I'll go ahead and finish it up and tie it into the rest of the game.

**Comment by WasabiFan** [comment]
Just as a heads-up, I have added the work items that we discussed yesterday as GH issues and assigned them to the people who volunteered.

**Comment by WasabiFan** [comment]
Big heads-up (@kyranstar this probably affects you): I just merged some fairly large changes into the master branch. Any changes in any of the map files will probably result in merge conflicts, so pull now before making more changes.

My updates add a "hints" section on the side of the window. In the future it will be hidden until you activate it, but for now it covers 1/5 of the window and hides some game content.

**Comment by WasabiFan** [comment]
Does anyone else have ideas for tasks that need to be completed before the 28th? If so, open a new issue for it so that we can track it. Think as much as you can about this because I don't want to be caught off-guard on the day before the demo when we discover that we haven't met some vital part of the spec.
